<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsAttributeTableDialog</class>
 <widget class="QDialog" name="QgsAttributeTableDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1054</width>
    <height>570</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Attribute Table</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <property name="spacing">
    <number>3</number>
   </property>
   <item row="6" column="0">
    <widget class="QgsDualView" name="mMainView">
     <property name="currentIndex">
      <number>-1</number>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QFrame" name="mUpdateExpressionBox">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <property name="spacing">
       <number>3</number>
      </property>
      <property name="leftMargin">
       <number>3</number>
      </property>
      <property name="topMargin">
       <number>1</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>3</number>
      </property>
      <item>
       <widget class="QgsFieldComboBox" name="mFieldCombo"/>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>=</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QgsFieldExpressionWidget" name="mUpdateExpressionText">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="mRunFieldCalc">
        <property name="text">
         <string>Update All</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="mRunFieldCalcSelected">
        <property name="text">
         <string>Update Selected</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="8" column="0">
    <layout class="QHBoxLayout">
     <property name="leftMargin">
      <number>3</number>
     </property>
     <property name="topMargin">
      <number>3</number>
     </property>
     <property name="rightMargin">
      <number>3</number>
     </property>
     <property name="bottomMargin">
      <number>3</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <widget class="QgsFeatureFilterWidget" name="mFeatureFilterWidget" native="true"/>
       </item>
       <item>
        <widget class="QToolButton" name="mAttributeViewButton">
         <property name="toolTip">
          <string>Switch to form view</string>
         </property>
         <property name="text">
          <string>Form View</string>
         </property>
         <property name="icon">
          <iconset resource="../../images/images.qrc">
           <normaloff>:/images/themes/default/mActionFormView.svg</normaloff>:/images/themes/default/mActionFormView.svg</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">mMainViewButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="mTableViewButton">
         <property name="toolTip">
          <string>Switch to table view</string>
         </property>
         <property name="text">
          <string>Table View</string>
         </property>
         <property name="icon">
          <iconset resource="../../images/images.qrc">
           <normaloff>:/images/themes/default/mActionOpenTable.svg</normaloff>:/images/themes/default/mActionOpenTable.svg</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">mMainViewButtonGroup</string>
         </attribute>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item row="4" column="0">
    <widget class="QToolBar" name="mToolbar">
     <property name="iconSize">
      <size>
       <width>18</width>
       <height>18</height>
      </size>
     </property>
     <property name="floatable">
      <bool>false</bool>
     </property>
     <addaction name="mActionToggleEditing"/>
     <addaction name="mActionToggleMultiEdit"/>
     <addaction name="mActionSaveEdits"/>
     <addaction name="mActionReload"/>
     <addaction name="separator"/>
     <addaction name="mActionAddFeature"/>
     <addaction name="mActionDeleteSelected"/>
     <addaction name="mActionCutSelectedRows"/>
     <addaction name="mActionCopySelectedRows"/>
     <addaction name="mActionPasteFeatures"/>
     <addaction name="separator"/>
     <addaction name="mActionExpressionSelect"/>
     <addaction name="mActionSelectAll"/>
     <addaction name="mActionInvertSelection"/>
     <addaction name="mActionRemoveSelection"/>
     <addaction name="mActionSearchForm"/>
     <addaction name="mActionSelectedToTop"/>
     <addaction name="mActionPanMapToSelectedRows"/>
     <addaction name="mActionZoomMapToSelectedRows"/>
     <addaction name="separator"/>
     <addaction name="mActionAddAttribute"/>
     <addaction name="mActionRemoveAttribute"/>
     <addaction name="mActionOpenFieldCalculator"/>
     <addaction name="separator"/>
     <addaction name="mActionSetStyles"/>
     <addaction name="separator"/>
     <addaction name="mActionDockUndock"/>
    </widget>
   </item>
  </layout>
  <action name="mActionSearchForm">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionFilter2.svg</normaloff>:/images/themes/default/mActionFilter2.svg</iconset>
   </property>
   <property name="text">
    <string>Select/filter features using form</string>
   </property>
   <property name="toolTip">
    <string>Select/filter features using form (Ctrl+F)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="mActionToggleEditing">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionToggleEditing.svg</normaloff>:/images/themes/default/mActionToggleEditing.svg</iconset>
   </property>
   <property name="text">
    <string>Toggle editing mode</string>
   </property>
   <property name="toolTip">
    <string>Toggle editing mode (Ctrl+E)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="mActionToggleMultiEdit">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionMultiEdit.svg</normaloff>:/images/themes/default/mActionMultiEdit.svg</iconset>
   </property>
   <property name="text">
    <string>Toggle multi edit mode</string>
   </property>
   <property name="toolTip">
    <string>Toggle multi edit mode</string>
   </property>
  </action>
  <action name="mActionSaveEdits">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionSaveAllEdits.svg</normaloff>:/images/themes/default/mActionSaveAllEdits.svg</iconset>
   </property>
   <property name="text">
    <string>Save edits</string>
   </property>
   <property name="toolTip">
    <string>Save edits (Ctrl+S)</string>
   </property>
  </action>
  <action name="mActionReload">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionRefresh.svg</normaloff>:/images/themes/default/mActionRefresh.svg</iconset>
   </property>
   <property name="text">
    <string>Reload the table</string>
   </property>
   <property name="toolTip">
    <string>Reload the table</string>
   </property>
  </action>
  <action name="mActionAddFeature">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionNewTableRow.svg</normaloff>:/images/themes/default/mActionNewTableRow.svg</iconset>
   </property>
   <property name="text">
    <string>Add feature</string>
   </property>
  </action>
  <action name="mActionDeleteSelected">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionDeleteSelectedFeatures.svg</normaloff>:/images/themes/default/mActionDeleteSelectedFeatures.svg</iconset>
   </property>
   <property name="text">
    <string>Delete selected features</string>
   </property>
   <property name="toolTip">
    <string>Delete selected features</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="mActionExpressionSelect">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mIconExpressionSelect.svg</normaloff>:/images/themes/default/mIconExpressionSelect.svg</iconset>
   </property>
   <property name="text">
    <string>Select features using an expression</string>
   </property>
   <property name="toolTip">
    <string>Select features using an expression</string>
   </property>
  </action>
  <action name="mActionSelectAll">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionSelectAll.svg</normaloff>:/images/themes/default/mActionSelectAll.svg</iconset>
   </property>
   <property name="text">
    <string>Select all</string>
   </property>
   <property name="toolTip">
    <string>Select all (Ctrl+A)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="mActionInvertSelection">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionInvertSelection.svg</normaloff>:/images/themes/default/mActionInvertSelection.svg</iconset>
   </property>
   <property name="text">
    <string>Invert selection</string>
   </property>
   <property name="toolTip">
    <string>Invert selection (Ctrl+R)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="mActionRemoveSelection">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionDeselectAll.svg</normaloff>:/images/themes/default/mActionDeselectAll.svg</iconset>
   </property>
   <property name="text">
    <string>Deselect all</string>
   </property>
   <property name="toolTip">
    <string>Deselect all (Ctrl+Shift+A)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+A</string>
   </property>
  </action>
  <action name="mActionSelectedToTop">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionSelectedToTop.svg</normaloff>:/images/themes/default/mActionSelectedToTop.svg</iconset>
   </property>
   <property name="text">
    <string>Move selection to top</string>
   </property>
   <property name="toolTip">
    <string>Move selection to top</string>
   </property>
  </action>
  <action name="mActionPanMapToSelectedRows">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionPanToSelected.svg</normaloff>:/images/themes/default/mActionPanToSelected.svg</iconset>
   </property>
   <property name="text">
    <string>Pan map to the selected rows</string>
   </property>
   <property name="toolTip">
    <string>Pan map to the selected rows (Ctrl+P)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="mActionZoomMapToSelectedRows">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionZoomToSelected.svg</normaloff>:/images/themes/default/mActionZoomToSelected.svg</iconset>
   </property>
   <property name="text">
    <string>Zoom map to the selected rows</string>
   </property>
   <property name="toolTip">
    <string>Zoom map to the selected rows (Ctrl+J)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+J</string>
   </property>
  </action>
  <action name="mActionCutSelectedRows">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionEditCut.svg</normaloff>:/images/themes/default/mActionEditCut.svg</iconset>
   </property>
   <property name="text">
    <string>Cut selected rows to clipboard</string>
   </property>
   <property name="toolTip">
    <string>Cut selected rows to clipboard (Ctrl+X)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="mActionCopySelectedRows">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionEditCopy.svg</normaloff>:/images/themes/default/mActionEditCopy.svg</iconset>
   </property>
   <property name="text">
    <string>Copy selected rows to clipboard</string>
   </property>
   <property name="toolTip">
    <string>Copy selected rows to clipboard (Ctrl+C)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="mActionPasteFeatures">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionEditPaste.svg</normaloff>:/images/themes/default/mActionEditPaste.svg</iconset>
   </property>
   <property name="text">
    <string>Paste features from clipboard</string>
   </property>
   <property name="toolTip">
    <string>Paste features from clipboard (Ctrl+V)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="mActionRemoveAttribute">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionDeleteAttribute.svg</normaloff>:/images/themes/default/mActionDeleteAttribute.svg</iconset>
   </property>
   <property name="text">
    <string>Delete field</string>
   </property>
   <property name="toolTip">
    <string>Delete field (Ctrl+L)</string>
   </property>
  </action>
  <action name="mActionAddAttribute">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionNewAttribute.svg</normaloff>:/images/themes/default/mActionNewAttribute.svg</iconset>
   </property>
   <property name="text">
    <string>New field</string>
   </property>
   <property name="toolTip">
    <string>New field (Ctrl+W)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="mActionOpenFieldCalculator">
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionCalculateField.svg</normaloff>:/images/themes/default/mActionCalculateField.svg</iconset>
   </property>
   <property name="text">
    <string>Open field calculator</string>
   </property>
   <property name="toolTip">
    <string>Open field calculator (Ctrl+I)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="mActionSetStyles">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionConditionalFormatting.svg</normaloff>:/images/themes/default/mActionConditionalFormatting.svg</iconset>
   </property>
   <property name="text">
    <string>Conditional formatting</string>
   </property>
   <property name="toolTip">
    <string>Conditional formatting</string>
   </property>
  </action>
  <action name="mActionDockUndock">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../images/images.qrc">
     <normaloff>:/images/themes/default/mDockify.svg</normaloff>:/images/themes/default/mDockify.svg</iconset>
   </property>
   <property name="text">
    <string>Dock Attribute Table</string>
   </property>
   <property name="toolTip">
    <string>Dock Attribute Table</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsFieldComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsfieldcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsFieldExpressionWidget</class>
   <extends>QWidget</extends>
   <header>qgsfieldexpressionwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsDualView</class>
   <extends>QStackedWidget</extends>
   <header>qgsdualview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsFeatureFilterWidget</class>
   <extends>QWidget</extends>
   <header>qgsfeaturefilterwidget_p.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>mFieldCombo</tabstop>
  <tabstop>mUpdateExpressionText</tabstop>
  <tabstop>mRunFieldCalc</tabstop>
  <tabstop>mRunFieldCalcSelected</tabstop>
  <tabstop>mAttributeViewButton</tabstop>
  <tabstop>mTableViewButton</tabstop>
 </tabstops>
 <resources>
  <include location="../../images/images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>mMainViewButtonGroup</sender>
   <signal>buttonClicked(int)</signal>
   <receiver>mMainView</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>393</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mActionToggleEditing</sender>
   <signal>toggled(bool)</signal>
   <receiver>mActionSaveEdits</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mActionToggleEditing</sender>
   <signal>toggled(bool)</signal>
   <receiver>mActionReload</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="mMainViewButtonGroup"/>
 </buttongroups>
</ui>
